src/lib.rs

```diff
@@ -835,6 +835,27 @@ impl<T: TensorType> Tensor<T> {
         }
     }
 
+    /// Sets (copies) the tensor values to the provided ones.
+    /// 
+    /// ```
+    /// # use tensorflow::Tensor;
+    /// let a = Tensor::new(&[2, 2]).with_values(&[0_i32, 1, 2, 3]);
+    /// ```
+    ///
+    /// # Panic
+    /// If `value.len() != self.len()`.
+    pub fn with_values(mut self, value: &[T]) -> Self {
+        assert!(
```

Comment:
Bad user input shouldn't produce panics (if we can avoid it).  This should return a `Result`, instead.  You could use `invalid_arg!` here.  Callers can always call `unwrap()` if they're sure they did the right thing.